index.d.ts

```diff
@@ -0,0 +1,33 @@
+declare class MapExpander {
+  expandMap(currentMap: Object, pathString: string, mapValue: Object): Object;
+  getMapValue(currentMap: Object, pathString: string): Object;
+}
+
+declare class Response {
+  constructor(body: Object, statusCode: number, headers: Object): void;
+}
+
+declare class ApiFactory {
+  constructor(options: object, overrideLogger: object);
```

Comment:
We can provide explicit `optionsObject` class to contain the properties which it can have.